js/translation-utils.js

```diff
@@ -0,0 +1,157 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+'use strict';
+const path = require('path');
+const jsonUtils = require('./jsonUtils.js');
+const glob = require('glob');
+const zluxUtil = require('./util');
+const acceptLanguageParser = require('accept-language-parser');
+
+const utilLog = zluxUtil.loggers.utilLogger;
+
+/* loadTranslations reads pluginDefinition.i18n.{lang}-{Country}.json files
+/  for a given plugin location.
+/ 
+/  Returns translation maps:
+/  {
+/    '{lang}-{Country}': {
+/      'key': 'value',
+/      ...
+/    },
+/    ...
+/  }
+*/
+function loadTranslations(pluginLocation) {
+  const translationMaps = {};
+  const relativePath = 'web/assets/i18n';
+  const filePrefix = 'pluginDefinition.i18n.';
+  const fileExt = '.json';
+  const pattern = path.join(
+    pluginLocation,
+    relativePath,
+    `${filePrefix}*${fileExt}`
+  );
+  const files = glob.sync(pattern, {});
```

Comment:
For the record: yes this is a synchronous call but no it shouldn't really be bad since this function is done one per plugin at startup. I feel that we have not captured all scenarios in which we will load plugins after startup, but it seems like we'd want to avoid loading plugins at scale, such that this again shouldn't impact performance.